ui/src/funding/contract.ts

```diff
@@ -0,0 +1,92 @@
+import type { BigNumberish, ContractTransaction, Signer } from "ethers";
+import {
+  Erc20Pool,
+  Erc20Pool__factory as PoolFactory,
+  ERC20,
+  ERC20__factory as Erc20Factory,
+} from "radicle-contracts/build/contract-bindings/ethers";
+
+// TODO(nuno): make the contract addresses configurable/env-dependant
+
+export const POOL_ADDRESS: string =
+  "0x37c10d847bf9e708add451bf2f80c1297d7aa691";
+
+export function pool(
+  signer: Signer,
+  address: string = POOL_ADDRESS
+): PoolContract {
+  return new PoolContract(signer, address);
+}
+
+export const ERC20_TOKEN_ADDRESS = "0xf34a89802590f944e3de71b1f74d66ed1bafc9cd";
+
+export function erc20Token(signer: Signer): ERC20 {
+  return Erc20Factory.connect(ERC20_TOKEN_ADDRESS, signer);
+}
+
+// PoolContract is a wrapper type around the actual contract, `Erc20Pool`,
+// that offers a more ergonomic API around the contract.
+export class PoolContract {
+  contract: Erc20Pool;
+
+  constructor(signer: Signer, address: string = POOL_ADDRESS) {
```

Comment:
It seems to me that we never call this constructor with just one argument. The `pool` function handles the default value. This means we can just remove the default.